workflows_core/workflow/context_manager.py

```diff
@@ -100,9 +100,15 @@ def __exit__(self, exc_type: type, exc_value: BaseException, traceback: Tracebac
                     workflow_name=self._workflow_name,
                 )
             else:
-                self._set_status(
-                    status=self.COMPLETE, worker_number=self._engine.worker_number
-                )
+                if self._engine.output_to_status:
```

Comment:
given that if output_documents defaults to None, we shouldn't need an `if` statement here